crates/bevy_pbr/src/render/shadow_sampling.wgsl

```diff
@@ -0,0 +1,134 @@
+#define_import_path bevy_pbr::shadow_sampling
+
+#import bevy_pbr::mesh_view_bindings as view_bindings
+#import bevy_pbr::utils PI
+
+fn sample_shadow_map_hardware(light_local: vec2<f32>, depth: f32, array_index: i32) -> f32 {
+    // Do the lookup, using HW 2x2 PCF and comparison
+    // NOTE: Due to non-uniform control flow above, we must use the level variant of the texture
+    // sampler to avoid use of implicit derivatives causing possible undefined behavior.
+#ifdef NO_ARRAY_TEXTURES_SUPPORT
+    return textureSampleCompare(
```

Comment:
I think this should still be the `Level` variant even if naga isn't emitting validation errors. At least, for sampling shadow maps that anyway only have one mip level (0) it is what _should_ happen, but it's nicer to be explicit about it.
```suggestion
    return textureSampleCompareLevel(
```